validator/src/bootstrap.rs

```diff
@@ -1111,29 +1111,67 @@ mod with_incremental_snapshots {
         bootstrap_config: &RpcBootstrapConfig,
         rpc_peers: &[ContactInfo],
     ) -> Vec<PeerSnapshotHash> {
-        let trusted_snapshot_hashes = get_trusted_snapshot_hashes(cluster_info, validator_config);
+        // Which strategy to use for getting the peer snapshot hashes?  The standard way is what's
+        // described in the function's documentation.  However, if there are no trusted peers that
+        // have enabled incremental snapshots, it may be possible that there are no snapshot hashes
+        // with a slot that is a multiple of our full snapshot archive interval.  If that happens,
+        // we retry with the "fallback" strategy that *does not* filter based on full snapshot
+        // archive interval.
+        #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+        enum Strategy {
+            Standard,
+            Fallback,
```

Comment:
this is a little more wordy but also avoids needing to define what `Fallback` means:
```suggestion
            WithoutFullSnapshotArchiveIntervalFiltering,
```

Reply:
Yeah, I debating doing something like this. I'll make this change.